Plugins/PluginsShared/PluginError.swift

```diff
@@ -0,0 +1,125 @@
+import PackagePlugin
+import Foundation
+
+enum PluginError: Swift.Error, CustomStringConvertible, LocalizedError {
+    case incompatibleTarget(targetName: String)
+    case noTargetOrDependenciesWithExpectedFiles(targetName: String, dependencyNames: [String])
+    case badArguments([String])
+    case noTargetsMatchingTargetName(targetName: String)
+    case tooManyTargetsMatchingTargetName(targetName: String, matchingTargetNames: [String])
+    case fileErrors([FileError])
+
+    var description: String {
+        switch self {
+        case .incompatibleTarget(let targetName):
+            return "Incompatible target called '\(targetName)'. Only Swift source targets can be used with the Swift OpenAPI Generator plugin."
+        case .noTargetOrDependenciesWithExpectedFiles(let targetName, let dependencyNames):
+            let introduction = dependencyNames.isEmpty ? "Target called '\(targetName)' doesn't contain" : "Target called '\(targetName)' or its local dependencies \(dependencyNames) don't contain"
+            return "\(introduction) any config or document files with expected names. For OpenAPI code generation, a target needs to contain a config file named 'openapi-generator-config.yaml' or 'openapi-generator-config.yml', as well as an OpenAPI document named 'openapi.yaml', 'openapi.yml' or 'openapi.json' under target's source directory. See documentation for details."
+        case .badArguments(let arguments):
+            return "On Xcode, use Xcode's command plugin UI to choose one specific target before hitting 'Run'. On CLI make sure arguments are exactly of form '--target <target-name>'. The reason for this error is unexpected arguments: \(arguments)"
```

Comment:
Adopters might be using this from other IDEs, so I'd drop the reference to Xcode specifically.
```suggestion
            return "Unexpected arguments: '\(arguments.joined(separator: " "))', expected: '--target MyTarget'."
```

Reply:
> Xcode UI - unless they explicitly add extra arguments, they'll never see this error, as the Xcode UI will force them to select one product, and provide the right flags.

Not quite right. If they don't choose any target, Xcode will pass _all targets_ automatically in form of ["--target", "TARGET1", "--target", "TARGET2"]. If users press the "Run" button without choosing a product, this error will help them know what to do.
It's also possible for users to multi-select multiple products.

Knowing that, do you still want me to remove the reference to Xcode?

Reply:
Yes, let's remove the reference.

But the multiple --target case is interesting. Would it be difficult to just iterate them all, now that we already crawl local dependencies anyway? That way, if you don't provide an explicit target, we run on them all?

Reply:
Not really, shouldn't be difficult maybe very slightly worse error handling, but not sure, might be worth it. Should i go for it?

Reply:
Yeah I guess so. If we can't have full script mode, where the user provides exactly one target that we run it on, with great error handling, we might as well go the other direction, and allow running on all targets.

Once we added the dependency crawling, we lost most of the great error handling anyway, so this change doesn't seem as that big of a leap, and the experience for new adopters should be nicer too.